src/Workspaces/Core/Portable/Workspace/Solution/SolutionChanges.cs

```diff
@@ -80,4 +83,24 @@ public IEnumerable<AnalyzerReference> GetRemovedAnalyzerReferences()
             }
         }
     }
+
+    internal IEnumerable<DocumentId> GetChangedFrozenSourceGeneratedDocuments()
+    {
+        if (_newSolution.CompilationState.FrozenSourceGeneratedDocumentStates is null)
+            return [];
```

Comment:
this ansewrs a question i previously had.  so it seems like we can remove people who check if this is null/non-null before calling GetChangedFrozenSourceGeneratedDocuments